SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1456,8 +1461,13 @@ bool CreateCustomAuthoritySet(FString SchemaInputPath, FString SchemaOutputPath,
 			return true;
 		});
 
+		// Put the generated schema binary for our custom schema in the Intermediate folder.
+		// This is so it doesn't stomp the users (or tests) proper schema binary used by the runtime.
+		FString SchemaCompilerIntermediateDir = FPaths::Combine(FPaths::ProjectIntermediateDir(), TEXT("Improbable"), TEXT("schema"));
+		Filesystem.CreateDirectoryTree(*SchemaCompilerIntermediateDir);
+
 		FString SchemaJsonPath;
-		if (!RunSchemaCompiler(SchemaJsonPath, SchemaInputPath))
+		if (!RunSchemaCompiler(SchemaJsonPath, SchemaInputPath, "" /*BuildDir*/, SchemaCompilerIntermediateDir))
```

Comment:
I assume this still breaks in the old way if you leave out the `SchemaCompilerIntermediateDir` parameter?

Reply:
Yeah, this is because the usage of the `SchemaCompiler` for `CreateServerWorkerAuthoritySet` is out of the ordinary. It creates a schema binary using some custom schema files to get the component set IDs out (or something). The expected usage of the function is to only be part of a full schema gen. If you do anything but a full schema gen and don't pass `SchemaCompilerIntermediateDir`, you will stomp.

Reply:
Can't we make this so people can't stomp? API yeet this possibility out of existence by only passing in a base directory and making all paths out of that?

Reply:
e.g.
`SchemaInputDir = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("schema"));`
take in a single input parameter that is the `SpatialGDKServicesConstants::SpatialOSDirectory` override part.

Reply:
It's really not an API that's exposed, very very internal. 
Also we can't use a single base dir as it's required to have links to the standard_library schema too. I did think of that but then encountered issues actually running the compiler.
Let's just keep this simple and use as is.

Reply:
ok fair enough, you have more context : )